gatsby/content/docs/e2ee_cross_signing.mdx

```diff
@@ -0,0 +1,1198 @@
+   ephemeral key pair with `var sas = olm.SAS();` and sends Bob the public key, via `m.key.verification.key`.
+7. Bob receives the public key from Alice and sends his own public key. Bob's device displays
+   the emoji / numbers for verification now.
+8. Alice receives Bob's public key, and can finally verify the commitment she saved earlier. If all
+   matches, Alice's device will now display the emoji / numbers for verification.
+9. If the emoji / numbers both match up, they will send each other an `m.key.verification.mac` with
+   the MAC'd information of the keys that should be verified.
+10. Finally, if all checks out, they send each other a `m.key.verification.done` and
+    the verification process is concluded.
+
+### Things to look out for
+1. *Always* verify that an incoming message has a valid, known transaction ID.
+2. *Always* verify that an incoming message is no more than 10 minutes in the past.
+3. *Always* stop the verification process as soon as you receive a `m.key.verification.cancel`, be
+   it for whatever reason.
+4. *Don't* just blindly assume that methods, `message_authentication_code`, etc. are as outlined here.
+   More might be added in the future (e.g. to fix a discovered security flaw), as such *always* verify
+   which ones you actually support and act accordingly.
+5. Don't forget to `sas.free();` your generated SAS object all at the end of verification or when timing
+   out or when it is being canceled.
+6. Don't forget to calculate *and verify* the commitment!
+7. It might be a good idea to generalize the code handling the verification flow so that you can easily
+   add other methods than `m.sas.v1` later on.
+
+## 2. Display verification status based on signatures
+Don't worry, after having implemented SAS this should be a breeze, it's way simpler and way less of a
+hassle to implement!
+
+### General ideas of Cross-Signing
+The general idea of cross-signing is that instead of every device having to verify every other device,
+people will only have to verify other people, and you will have to verify each of your own new sessions
+(logins) only once. For this, each *user* has three keys: A master key, a self-signing key and a
+user-signing key. The master key will sign both the self-signing and the user-signing key. The
+self-signing key will then sign your own device keys, and the user-signing key will sign other people's
+master keys. The purpose of splitting this up into three keys is that in the case of either the
+self-signing key or the user-signing key getting compromised, you can easily exchange them, while
+still keeping the master key. As the master key is *only* used for signing your *own* user-signing and
+self-signing key, it is rarely used, so the attack surface to obtain it is minimal. Optionally, device
+keys themselves can sign their own master key.
+
+![](./matrix-cross-signing-key-diagram.svg)
```

Comment:
The image itself needs to be moved to `./gatsby/static/docs/guides/img`, and then this link can be updated to `/docs/guides/img/matrix-cross-signing-key-diagram.svg`.